README.md

```diff
@@ -1,85 +1,711 @@
 # Toggle Form
 
-A simple form library using react hooks
-
-
-## Features
-
-- Powerful validation
-- Supports nested forms
-- Supports conditional fields
-- Reasonably typesafe
-
-## Introduction
-
-The library exposes three react hooks to define a form.
-
+A simple form library using react hooks.
+# Getting started
+This page is an overview of the `Toggle Form` documentation and related resources. <br>
+The library exposes three react hooks to define a form:
 1. useForm
 2. useArrayField
 3. useObjectField
 
-### useForm
-
-`useForm` hook is used to control a form.
-The `useForm` hook accepts these options:
-
-|option|description|
-|----|----|
-|value|The initial value for the form|
-|schema|The schema used to validate the form|
-
-The return value from `useForm` hooks are as follows:
-
-|value|description|
-|----|----|
-|value|The value of the form|
-|error|The error state of the form|
-|pristine|The pristine state of the form|
-|pristine|Defines if the form is pristine|
-|validate|If there are errors, sets form error else returns sanitized value|
-|setPristine|Function to set form's pristine state|
-|setError|Function to set form's error state|
-|setValue|Function to set form's value, error is cleared and pristine is set to true|
-|setFieldValue|Function to set form field's value and pristine is set to false|
-
-### useFormObject
-
-`useFormObject` hook is used to control an object.
-The `useFormObject` hook accepts these options:
-
-|option|description|
-|----|----|
-|name|name of the object field|
-|onChange| change handler of the object|
-|defaultValue|default value for the object|
-
-The return value from `useFormObject` is the change handler for the object
-fields.
-
-### useFormArray
 
-`useFormArray` hook is used to control an array.
-The `useFormArray` hook accepts these options:
+## Prerequisites
+- [Node](https://nodejs.org/en/)
+- Understanding of [TypeScript](https://www.typescriptlang.org/)
 
-|option|description|
-|----|----|
-|name|name of the array field|
-|onChange| change handler of the array|
+---
 
-The return value from `useFormArray` are as follows:
+## Installation 
+```
+yarn install 
 
-|value|description|
-|----|----|
-|setValue|The change handler for the array items.|
-|removeValue|The delete handler for the array items.|
+yarn add @togglecorp/toggle-form@beta 
+```
+---
+
+## Table of Contents
+  - [Quick start](#quick-start)
+  - [useForm](#useform)
+    - [setValue](#setvalue)
+    - [setFieldValue](#setfieldvalue)
+    - [setPristine](#setpristine)
+    - [setError](#seterror)
+  - [getObjectError](#getobjecterror)
+  - [createSubmitHandler](#createsubmithandler)
+  - [useFormArray](#useformarray)
+  - [useFormObject](#useformobject)
+  - [Schema](#schema)
+
+---
+## Note:
+
+  In this documentation, we will be using the `RawInput` component instead of `input`.
```

Comment:
Any reason to add "instead of 'input'"?
We cannot use input directly with toggle form.

Reply:
```suggestion
  We will be using the `RawInput` component instead of `input`.
```